core/src/main/java/com/datastax/oss/driver/api/core/config/ProgrammaticDriverConfigLoader.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.core.config;
+
+import com.datastax.oss.driver.internal.core.config.typesafe.DefaultDriverConfigLoader;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import com.typesafe.config.ConfigValueFactory;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public class ProgrammaticDriverConfigLoader {
+
+  private ProgrammaticDriverConfigLoader() {}
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static class Builder {
+
+    private Config config = ConfigFactory.empty();
+
+    private Supplier<Config> fallbackSupplier = DefaultDriverConfigLoader.DEFAULT_CONFIG_SUPPLIER;
+
+    private Builder() {}
+
+    public Builder withBoolean(DriverOption option, boolean value) {
+      return with(option, value);
+    }
+
+    public Builder withBoolean(String profileName, DriverOption option, boolean value) {
```

Comment:
One idea I had for profiles which would be feasible would be to have a way for user a to create a `DriverConfigProfile` and then have a `withProfile(DriverConfigProfile profile)` method in here.  That would reduce the duplicated methods.   The one downside of that is that you could not use these`DriverConfigProfile` objects when working with the driver, but we could probably find a way to throw an early runtime exception if someone tries that.    What do you think of this idea?

Reply:
Mmm not a fan...
How about a "sub-builder":
```java
withProfile(
    "slow",
    profileBuilder()
        .withDuration(DefaultDriverOption.REQUEST_TIMEOUT, Duration.ofSeconds(30))
        .build()
)
```
It could reuse the same internal interface that I'm suggesting below.